pkg/build/api/v1/conversion.go

```diff
@@ -166,17 +166,27 @@ func convert_api_BuildSource_To_v1_BuildSource(in *newer.BuildSource, out *Build
 	if err := s.DefaultConvert(in, out, conversion.IgnoreMissingFields); err != nil {
 		return err
 	}
+	typeFound := false
 	switch {
 	// it is legal for a buildsource to have both a git+dockerfile source, but in v1 that was represented
 	// as type git.
 	case in.Git != nil:
 		out.Type = BuildSourceGit
+		typeFound = true
 	// it is legal for a buildsource to have both a binary+dockerfile source, but in v1 that was represented
 	// as type binary.
 	case in.Binary != nil:
 		out.Type = BuildSourceBinary
+		typeFound = true
 	case in.Dockerfile != nil:
 		out.Type = BuildSourceDockerfile
+		typeFound = true
+	case len(in.Images) > 0:
+		out.Type = BuildSourceImage
+		typeFound = true
+	}
+	if !typeFound {
+		out.Type = BuildSourceNone
```

Comment:
ok .. .i thought that earlier comment was an either/or regarding images, but yeah, i see the point now ... change pending.